src/hooks/useSortByTopic.ts

```diff
@@ -0,0 +1,28 @@
+import { useMemo } from 'react'
+import useSafeState from './useSafeState'
+
+interface SortTopic {
+  topic: string
+  order: boolean
+}
+
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+function useSortByTopic<T>(
+  data: T[],
+  defaultTopic: string,
+): { sortTopic: SortTopic; setSortTopic: React.Dispatch<React.SetStateAction<SortTopic>> } {
+  // true == 'asc' && false == 'dsc'
```

Comment:
Adding to Anxo's comment above, renaming `order` to `asc` makes it clearer. No need for the comment.